subprojects/docs/src/snippets/multiproject/dependencies-outgoingArtifact/groovy/producer/build.gradle

```diff
@@ -1,10 +1,16 @@
+plugins {
+    id 'java-library'
+}
+
+version = '1.0'
+
 task buildInfo(type: BuildInfo) {
-    version = project.version
-    outputFile = file("$buildDir/generated-resources/build-info.properties")
+    version.set(project.version)
+    outputFile.set(file("$buildDir/generated-resources/build-info.properties"))
 }
 
 sourceSets {
     main {
-        output.dir(buildInfo.outputFile.parentFile, builtBy: buildInfo)
+        output.dir(buildInfo.outputFile.getAsFile().get().parentFile, builtBy: buildInfo)
```

Comment:
We could use `buildInfo .map` on the task instead of `buildBy`. I think that's the "better" API.

Like here:
https://github.com/gradle/gradle/blob/cc2847f2c873c16d83ec086e23cd703fd093e2a5/buildSrc/subprojects/packaging/src/main/kotlin/gradlebuild/api-metadata.gradle.kts#L37

Reply:
Nice. Wasn't aware of this. I also updated the documentation section that uses this snippet, but am not confident with the wording. Please double check if this makes sense.

Reply:
The wording sounds good to me 👍